Source/Plugins/Core/com.equella.core/src/com/tle/integration/lti/generic/GenericLtiWrapperExtension.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to The Apereo Foundation under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * The Apereo Foundation licenses this file to you under the Apache License,
+ * Version 2.0, (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.tle.integration.lti.generic;
+
+import com.tle.common.externaltools.constants.ExternalToolConstants;
+import com.tle.core.guice.Bind;
+import com.tle.web.lti.usermanagement.LtiWrapperExtension;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpServletRequest;
+import org.apache.commons.lang.StringUtils;
+
+@SuppressWarnings("nls")
+@Bind
+@Singleton
+public class GenericLtiWrapperExtension implements LtiWrapperExtension {
+  @Override
+  public String getUserId(HttpServletRequest request) {
+    return null;
+  }
+
+  @Override
+  public String getUserId(HttpServletRequest request, String param) {
+    if (StringUtils.isNotEmpty(param)) {
+      if (StringUtils.isNotEmpty(
```

Comment:
Since we are in the Generic LTI logic, can we remove the check for productFamilyCode? My understanding is now that we have the configuration at the LTI Consumer level, this code should not care which LTI tool is calling oEQ.  If the canvas or d2l flows are still required, users have a separate lti signon URL for that (for now).

Reply:
Already answered.